javascript/ql/src/experimental/heuristics/ql/src/Security/CWE-807/ConditionalBypass.ql

```diff
@@ -0,0 +1,120 @@
+/**
+ * @name User-controlled bypass of security check with additional heuristic sources
+ * @description Conditions that the user controls are not suited for making security-related decisions.
+ * @kind path-problem
+ * @problem.severity error
+ * @security-severity 7.8
+ * @precision medium
+ * @id js/user-controlled-bypass-more-sources
+ * @tags experimental
+ *       security
+ *       external/cwe/cwe-807
+ *       external/cwe/cwe-290
+ */
+
+import javascript
+import semmle.javascript.security.dataflow.ConditionalBypassQuery
+import DataFlow::PathGraph
+
+/**
+ * Holds if the value of `nd` flows into `guard`.
```

Comment:
This is the only query in the heuristics folder with non-trivial contents. Would it be possible to move the `predicates` in a shared file?

Reply:
Yes, and there already is a `ConditionalBypassQuery.qll` file where these predicates should have been.